src/execution_profile.rs

```diff
@@ -0,0 +1,73 @@
+use pyo3::prelude::*;
+use scylla::client;
+use std::time::Duration;
+
+use crate::enums::{Consistency, SerialConsistency};
+
+#[pyclass(frozen)]
+#[derive(Clone)]
+pub(crate) struct ExecutionProfile {
+    pub(crate) _inner: client::execution_profile::ExecutionProfile,
+}
+
+#[pymethods]
+impl ExecutionProfile {
+    #[new]
+    #[pyo3(signature = (timeout=None, consistency=None, serial_consistency=None))]
+    pub(crate) fn new(
+        timeout: Option<PyObject>,
+        consistency: Option<PyObject>,
+        serial_consistency: Option<PyObject>,
+    ) -> PyResult<Self> {
+        let mut profile_builder = client::execution_profile::ExecutionProfile::builder();
+
+        if let Some(timeout) = timeout {
+            let secs = Python::with_gil(|py| timeout.extract::<f64>(py))?;
+            if !secs.is_finite() || secs <= 0.0 {
```

Comment:
Note: We really need a good solution for error handling. This `?` here is most likely not what we will ultimately want (but for this PR its ok). Intead we should give users an error that explains which parameter has wrong type.